model/src/main/java/org/projectnessie/model/Field.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.model;
+
+import java.util.List;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+@Value.Immutable(prehash = true)
+@JsonSerialize(as = ImmutableField.class)
+@JsonDeserialize(as = ImmutableField.class)
+public interface Field {
+
+  enum FieldType {
+
+  }
+
+  FieldType getType();
+
+  String getName();
+
+  List<Field> getChildren();
+
+}
```

Comment:
Agreed. Its not great to introduce this shim layer for the schema. This is primarily for on-the-wire transport as I would assume Nessie clients would convert from known schema to this on-the-wire format and we will translate to a known schema type for long term storage. To be clear we don't ascribe any meaning to these schemas or enforce them anywhere, its purely a transport mechanism. I would love to have anyones suggestions for a better way, though I don't want to couple our on-the-wire format and public api to a large library :disappointed:

Reply:
After a cursory glance at arrow it looks like the `vector` module would have to be added to get Arrow's schema. This I think is a rather heavy dependency. 

@sandhyasun what is the purpose of storing the schema at all? this data can always be recovered/cached on the engine right? cc @snazy

Reply:
I am currently testing out https://arrow.apache.org/docs/format/CDataInterface.html as the on the wire format. this would avoid us creating a whole structure ourselves and the `Schema` would then be a `Map<String, String>`. The schema is simple enough that we wouldn't need to pull any arrow libraries in to parse it, What do you guys think?

Reply:
The schema is needed by most engines for views especially, since it would make information schema queries impossibly slow if we have to expand each view dynamically to retrieve and build the schema.  But if we can come up with a scheme to suggest how one can map a version of a view with the schema and suggest caching it in  each engines cache , it may be fine. Since it's best not to store the hash of the commits outside of Nessie, we need to come up with another way to map a version in Nessie uniquely - maybe "change id"+id combination ?

Reply:
If we are going with a String format, can't we jsut use the Json string format of Arrow schema ? There is a [method](https://arrow.apache.org/docs/java/reference/org/apache/arrow/vector/types/pojo/Schema.html#toJson--) to convert Arrow to String which we use to pass into Nessie ?